app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -84,9 +84,15 @@ def link_identity_to_service_provider
         current_user: current_user,
         ial: resolved_authn_context_int_ial,
         rails_session_id: session.id,
+        email_address_id: email_address_id,
       )
     end
 
+    def email_address_id
+      return nil unless IdentityConfig.store.feature_select_email_to_share_enabled
+      return session[:selected_email_id] if session[:selected_email_id].present?
```

Comment:
We should probably be using `user_session` here, so it's destroyed when the user signs out.